packages/babel-helper-create-class-features-plugin/src/features.js

```diff
@@ -18,49 +24,67 @@ export const FEATURES = Object.freeze({
 const featuresKey = "@babel/plugin-class-features/featuresKey";
 const looseKey = "@babel/plugin-class-features/looseKey";
 
+// See https://github.com/babel/babel/issues/11622.
+// Since preset-env sets loose for the fields and private methods plugins, it can
+// cause conflicts with the loose mode set by an explicit plugin in the config.
+// To solve this problem, we ignore preset-env's loose mode if another plugin
+// explicitly sets it
+// The code to handle this logic doesn't check that "low priority loose" is always
+// the same. However, it is only set by the preset and not directly by users:
+// unless someone _wants_ to break it, it shouldn't be a problem.
+const looseLowPriorityKey =
+  "@babel/plugin-class-features/looseLowPriorityKey/#__internal__@babel/preset-env__please-overwrite-loose-instead-of-throwing";
+
 export function enableFeature(file, feature, loose) {
   // We can't blindly enable the feature because, if it was already set,
   // "loose" can't be changed, so that
   //   @babel/plugin-class-properties { loose: true }
   //   @babel/plugin-class-properties { loose: false }
   // is transformed in loose mode.
   // We only enabled the feature if it was previously disabled.
-  if (!hasFeature(file, feature)) {
+  if (!hasFeature(file, feature) || canIgnoreLoose(file, feature)) {
     file.set(featuresKey, file.get(featuresKey) | feature);
-    if (loose) file.set(looseKey, file.get(looseKey) | feature);
+    if (
+      loose ===
+      "#__internal__@babel/preset-env__prefer-true-but-false-is-ok-if-it-prevents-an-error"
```

Comment:
Lol I can't prevent myself from thinking of https://github.com/facebook/react/blame/80bff5397bf854750dbe7c286f61654ea58938c5/src/umd/ReactUMDEntry.js#L21